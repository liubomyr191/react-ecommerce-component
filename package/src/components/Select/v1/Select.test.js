import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import { testInput } from "composable-form-tests";
import Select from "./Select";

const OPTIONS = [
  { label: "A", value: "a" },
  { label: "B", value: "b" },
  { label: "C", value: "c" }
];

function simulateChanged(wrapper, value) {
  const labelToSelect = value.toString().toUpperCase();
  const selectOption = wrapper
    .find("div.react-select__option")
    .findWhere((n) => (n.props().children || "").trim() === labelToSelect);
  wrapper.setState({ focusedOption: undefined });

  selectOption.simulate("click");
  wrapper.update();
}

testInput({
  component: Select,
  defaultValue: null,
  exampleValueOne: "a",
  exampleValueTwo: "b",
  mount,
  options: OPTIONS,
  props: {
    menuIsOpen: true
  },
  simulateChanged
});

test("basic snapshot", () => {
  const component = renderer.create(<Select />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
