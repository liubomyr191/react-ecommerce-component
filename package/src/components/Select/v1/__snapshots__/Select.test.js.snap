// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alphabetize option snapshot 1`] = `
<div
  className="css-15mpz6f"
  id={undefined}
  onKeyDown={[Function]}
>
  <div
    className="css-1c6kyk8"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="css-1rtrksz"
    >
      <div
        className="css-vq28hp"
      >
        Select...
      </div>
      <input
        className="css-14uuagi"
        id="react-select-10-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        tabIndex="0"
        value=""
      />
    </div>
    <div
      className="css-1wy0on6"
    >
      <div
        aria-hidden="true"
        className="css-1lm7jx8"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <svg
          aria-hidden="true"
          className="css-19bqh2r"
          focusable="false"
          height={20}
          viewBox="0 0 20 20"
          width={20}
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div>
    <div
      className="css-z8mphd"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="css-1x0y3y4"
      >
        <div
          className="css-1gnokty"
          id="react-select-10-option-0"
          onClick={[Function]}
          onMouseMove={[Function]}
          onMouseOver={[Function]}
          role="option"
          tabIndex={-1}
        >
          A
        </div>
        <div
          className="css-1gnokty"
          id="react-select-10-option-1"
          onClick={[Function]}
          onMouseMove={[Function]}
          onMouseOver={[Function]}
          role="option"
          tabIndex={-1}
        >
          C
        </div>
        <div
          className="css-1gnokty"
          id="react-select-10-option-2"
          onClick={[Function]}
          onMouseMove={[Function]}
          onMouseOver={[Function]}
          role="option"
          tabIndex={-1}
        >
          E
        </div>
        <div
          className="css-1gnokty"
          id="react-select-10-option-3"
          onClick={[Function]}
          onMouseMove={[Function]}
          onMouseOver={[Function]}
          role="option"
          tabIndex={-1}
        >
          Z
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`basic snapshot 1`] = `
<div
  className="css-15mpz6f"
  id={undefined}
  onKeyDown={[Function]}
>
  <div
    className="css-1c6kyk8"
    onMouseDown={[Function]}
    onTouchEnd={[Function]}
  >
    <div
      className="css-1rtrksz"
    >
      <div
        className="css-vq28hp"
      >
        Select...
      </div>
      <input
        className="css-14uuagi"
        id="react-select-9-input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        tabIndex="0"
        value=""
      />
    </div>
    <div
      className="css-1wy0on6"
    >
      <div
        aria-hidden="true"
        className="css-1lm7jx8"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <svg
          aria-hidden="true"
          className="css-19bqh2r"
          focusable="false"
          height={20}
          viewBox="0 0 20 20"
          width={20}
        >
          <path
            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
          />
        </svg>
      </div>
    </div>
  </div>
  <div>
    <div
      className="css-z8mphd"
      onMouseDown={[Function]}
      onMouseMove={[Function]}
    >
      <div
        className="css-1x0y3y4"
      >
        <div
          className="css-1gnokty"
          id="react-select-9-option-0"
          onClick={[Function]}
          onMouseMove={[Function]}
          onMouseOver={[Function]}
          role="option"
          tabIndex={-1}
        >
          A
        </div>
        <div
          className="css-1gnokty"
          id="react-select-9-option-1"
          onClick={[Function]}
          onMouseMove={[Function]}
          onMouseOver={[Function]}
          role="option"
          tabIndex={-1}
        >
          B
        </div>
        <div
          className="css-1gnokty"
          id="react-select-9-option-2"
          onClick={[Function]}
          onMouseMove={[Function]}
          onMouseOver={[Function]}
          role="option"
          tabIndex={-1}
        >
          C
        </div>
      </div>
    </div>
  </div>
</div>
`;
